.github/workflows/build.yml

```diff
@@ -1,19 +1,27 @@
-name: Rust
+name: Build & Test
 
 on:
-  push:
+  push: 
+    branches: master
   pull_request:
+    branches: master
+  workflow_dispatch:
 
 env:
   CARGO_TERM_COLOR: always
 
+concurrency:
+  group: ublox-${{ github.workflow }}-${{ github.ref }}
```

Comment:
This is not an ideal concurrency group. If you merge an there's already a workflow run in progress on master, it will cancel it.

Reply:
Why not just `${{ github.workflow }}-${{ github.head_ref || github.run_id }}` ?

Reply:
I did not write this script, I don't understand how it got to this much complexity for just a few verifications.. If it were me, I would start from scratch and make it simple. There are two problems for me, I'm able to compile the default options while the CI fails. And I am unable to have either `ubx_proto27` or `ubx_proto31` pass on my side, while it apparently works on ""CI"".

Reply:
> Why not just `${{ github.workflow }}-${{ github.head_ref || github.run_id }}` ?

Whichever works better. My intent was to cancel current active runs when a PR gets update. True, it will have the same effect on master.  Your suggestion may work better. Feel free to update.

Reply:
> ${{ github.workflow }}-${{ github.head_ref || github.run_id }}

on PRs `github.head_ref` is set to the branch name, but when it's not a PR (like master branch) it will be `null` (since it's actually JSON) and so it will fallback to `github.run_id` and then the workflow run will be unique meaning it will never be cancelled by a another workflow run.

Reply:
> > ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
> 
> on PRs `github.head_ref` is set to the branch name, but when it's not a PR (like master branch) it will be `null` (since it's actually JSON) and so it will fallback to `github.run_id` and then the workflow run will be unique meaning it will never be cancelled by a another workflow run.

Awesome!  Learned something new today 😉